src/modules/contactUs/tests/contact-us.service.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { ContactService } from '../contact-us.service';
+import { ContactUsDto } from '../dto/contact-us.dto';
+import * as nodemailer from 'nodemailer';
+import { HttpException, HttpStatus } from '@nestjs/common';
+import * as validator from 'validator';
+
+jest.mock('nodemailer');
+
+describe('ContactService', () => {
+  let service: ContactService;
+  let sendMailMock: jest.Mock;
+
+  beforeEach(async () => {
+    sendMailMock = jest.fn().mockResolvedValue(true);
+    (nodemailer.createTransport as jest.Mock).mockReturnValue({
+      sendMail: sendMailMock,
+    });
+
+    // for testing purposes
+    process.env.CONTACT_EMAIL = 'contact@example.com';
+    process.env.EMAIL_USER = 'testuser@example.com';
+    process.env.EMAIL_PASS = 'password';
```

Comment:
What's happening here?

Reply:
The email configuration requires the use of environment variables. I tried using a different approach, but it didn't work.

Reply:
why are you defining constants like this though? I am confused

Reply:
I tried using direct constants for the test, but I kept getting an error saying 'email configuration is missing'. So, I had to use environment variables instead, which resolved the issue.

Reply:
updated